python/lsst/ip/diffim/utils.py

```diff
@@ -711,3 +732,354 @@ def plotWhisker(results, newWcs):
+        extent = (bbox.getBeginX(), bbox.getEndX(), bbox.getBeginY(), bbox.getEndY())
+        display2dArray(psfIm.getArray(), title='PSF', showBars=showBars, extent=extent)
+    return fig
+
+
+def displayCutouts(source, exposure, posImage=None, negImage=None, asHeavyFootprint=False, title=''):
+    """Use matplotlib.pyplot.imshow() to display cutouts within up to
+    three afw.image.Exposure's, given by an input SourceRecord.
+
+    @param source afw.table.SourceRecord defining the footprint to extract and display
+    @param exposure afw.image.Exposure from which to extract the cutout to display
+    @param posImage Second exposure from which to extract the cutout to display
+    @param negImage Third exposure from which to extract the cutout to display
+    @param asHeavyFootprint Display the cutouts as
+    afw.detection.HeavyFootprint, with regions outside the
+    footprint removed
+
+    @return matplotlib.pyplot.figure dispaying the image
+
+    """
+    plt = importMatplotlib()
+    if not plt:
+        return
+
+    fp = source.getFootprint()
+    bbox = fp.getBBox()
+    extent = (bbox.getBeginX(), bbox.getEndX(), bbox.getBeginY(), bbox.getEndY())
+
+    fig = plt.figure(figsize=(8, 2.5))
+    if not asHeavyFootprint:
+        subexp = afwImage.ImageF(exposure.getMaskedImage().getImage(), bbox, afwImage.PARENT)
+    else:
+        hfp = afwDet.HeavyFootprintF(fp, exposure.getMaskedImage())
+        subexp = getHeavyFootprintSubimage(hfp)
+    plt.subplot(1, 3, 1)
+    display2dArray(subexp.getArray(), title=title+' Diffim', extent=extent)
+    if posImage is not None:
+        if not asHeavyFootprint:
+            subexp = afwImage.ImageF(posImage.getMaskedImage().getImage(), bbox, afwImage.PARENT)
+        else:
+            hfp = afwDet.HeavyFootprintF(fp, posImage.getMaskedImage())
+            subexp = getHeavyFootprintSubimage(hfp)
+        plt.subplot(1, 3, 2)
+        display2dArray(subexp.getArray(), title=title+' Pos', extent=extent)
+    if negImage is not None:
+        if not asHeavyFootprint:
+            subexp = afwImage.ImageF(negImage.getMaskedImage().getImage(), bbox, afwImage.PARENT)
+        else:
+            hfp = afwDet.HeavyFootprintF(fp, negImage.getMaskedImage())
+            subexp = getHeavyFootprintSubimage(hfp)
+        plt.subplot(1, 3, 3)
+        display2dArray(subexp.getArray(), title=title+' Neg', extent=extent)
+    return fig
+
+
+def makeHeavyCatalog(catalog, exposure, verbose=False):
+    """Turn all footprints in a catalog into heavy footprints."""
+    for i, source in enumerate(catalog):
+        fp = source.getFootprint()
+        if not fp.isHeavy():
+            if verbose:
+                print(i, 'not heavy => heavy')
+            hfp = afwDet.HeavyFootprintF(fp, exposure.getMaskedImage())
+            source.setFootprint(hfp)
+
+    return catalog
+
+
+def getHeavyFootprintSubimage(fp, badfill=np.nan, grow=0):
+    """Extract the image from a heavyFootprint as an ImageF."""
```

Comment:
Done.  Also, moved this function out of `utils.py` and into `dipoleFitTask.DipoleFitAlgorithm` where it is used.